esque/controller/topic_controller.py

```diff
@@ -122,24 +124,29 @@ def update_from_cluster(self, topic: Topic):
         topic.is_only_local = False
 
         message = self._get_last_message(topic)
+        topic.last_message_timestamp = 0
         topic.last_message_timestamp = message.timestamp()[1]
 
         return topic
 
-    def _get_last_message(self, local_topic: Topic) -> Optional[Tuple[str, int]]:
-        high_watermark = local_topic.max_offset()["offset"] - 1
-        consumer = MessageConsumer(
-            PING_GROUP_ID,
-            local_topic.name,
-            True,
-            starting_offset=high_watermark,
-            partition=local_topic.max_offset()["partition_id"],
-        )
-        message = consumer.consume()
+    @translate_third_party_exceptions
+    def _get_last_message(self, local_topic: Topic) -> Message:
```

Comment:
How about adding `last_message_timestamp` to the Partition object. Then this could just be `max(partition.last_message_timestamp for partition in local_topic.partition_data)`.
If you then add it to `Partition.as_dict()` it will also work nicer with the standardized output we're about to introduce.

Reply:
No problem, I just thought that the issue was to show the last message for each topic, but adding it to the partition makes the code even easier, again ;)